### ISSUE TITLE
convert torch.tensor to numpy

### ISSUE THREAD

**Issue by ghost** [kevinzakka/recurrent-visual-attention#4]
Nice code. But
1. I think where you conver torch.tensor to numpy, where gradient backpropagation broken.
2. use_gpu do not works.

**Comment by kevinzakka** [comment]
Hey @0-oo-0, I'm aware of @2 and have not been able to fix it yet because I don't currently own a GPU and have finished my AWS credits. It's just a problem of applying `.cpu()` before `numpy()`.

As for number 1, the glimpse extraction operation is non-differentiable anyway, so there shouldn't be a need to back propagate the gradients to it. We want to back propagate the gradients up to the glimpse network which transforms the glimpse and the location vector into the glimpse feature vector `g_t`.

Correct me if I'm wrong though.

**Comment by 0-oo-0** [comment]
Yeah, I think it is fine. I have two sugestions and a question.
1. The experiment result in paper of final version is different from the arxiv version.
![default](https://user-images.githubusercontent.com/33923638/35784482-c56a4396-0a52-11e8-8c21-727a6299763d.png)
[https://papers.nips.cc/paper/5542-recurrent-models-of-visual-attention.pdf](url)
2. By replacing `F.tanh()` at https://github.com/kevinzakka/recurrent-visual-attention/blob/5a73041b18a2dc07b71193cfb55ae40be0eda42b/modules.py#L344 and https://github.com/kevinzakka/recurrent-visual-attention/blob/5a73041b18a2dc07b71193cfb55ae40be0eda42b/modules.py#L354 with `F.hardtanh()` , I get 1.12% error rate.
But at test time, `l_t = mu`, which suggested by [Torch Blog Post on RAM](torch.ch/blog/2015/09/21/rmva.html), results in a worse result. I do not konw why.

**Comment by kevinzakka** [comment]
@0-oo-0 hey, not sure I understand. What's the issue exactly? 

Your suggestion is that using `hard_tanh` reduces the test error further is that correct?

**Comment by 0-oo-0** [comment]
Yes, I update the comment.
